esphome/components/scd30/scd30.cpp

```diff
@@ -99,6 +109,8 @@ void SCD30Component::setup() {
     this->mark_failed();
     return;
   }
+
+  this->schedule_next_check_();
```

Comment:
Instead of continuously setting a new timeout, I would suggest using an interval instead:
```suggestion
  // check each 500ms if data is ready, and read it in that case
  this->set_interval("status-check", 500, [this]() {
    if (this->is_data_ready_())
      this->update();
  });
```